assets/js/admin/payment-method-restrictions.js

```diff
@@ -30,10 +23,31 @@ jQuery( function( $ ) {
 		var price = accounting.unformat( $( element ).val(), wcs_gateway_restrictions.decimal_point_separator );
 		price     = accounting.formatNumber( price, wcs_gateway_restrictions.number_of_decimal_places );
```

Comment:
Numbers over 999 get formated with a separator which was then breaking the parseFloat checks

```suggestion
		price     = accounting.formatNumber( price, wcs_gateway_restrictions.number_of_decimal_places, '' );
```